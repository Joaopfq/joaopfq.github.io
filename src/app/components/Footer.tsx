export function Footer() {
    return (
      <footer id="contato" className="text-center py-6 border-t mt-auto">
        <p>© {new Date().getFullYear()} João Paulo. Todos os direitos reservados.</p>
      </footer>
    );
  }
  