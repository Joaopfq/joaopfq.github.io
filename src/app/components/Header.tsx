export function Header() {
    return (
      <header className="bg-gray-900 p-4 flex justify-end items-center shadow-md">
        <nav className="space-x-4">
          <a href="#projetos" className="hover:underline">Projetos</a>
          <a href="#contato" className="hover:underline">Contato</a>
        </nav>
      </header>
    );
  }
  