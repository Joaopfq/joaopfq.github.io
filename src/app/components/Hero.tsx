export function Hero() {
    return (
      <section className="text-center py-16 px-4">
        <h3 className="text-2xl md:text-3xl font-bold mb-4">
          Olá! Eu sou João Paulo
        </h3>
        <p className="sm:text-base md:text-lg max-w-2xl mx-auto">
          Sou desenvolvedor Fullstack com experiência em  
          <b> Next</b>, <b>Vue</b>, <b>TypeScript</b>, <b>React</b>, <b>Java</b> e <b>Spring Boot</b>, 
          construindo aplicações completas com interfaces modernas, responsivas e acessíveis, além de backends escaláveis e bem estruturados.
        </p>
      </section>
    );
  }
  