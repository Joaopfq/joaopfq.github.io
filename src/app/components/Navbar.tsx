"use client";

import Link from "next/link";
import { usePathname } from "next/navigation";

const navLinks = [
  { href: "/projects", label: "Projetos" },
  { href: "/about", label: "Sobre" },
  { href: "#contact", label: "Contato" },
];

export function Navbar() {
  const pathname = usePathname();

  return (
    <nav className="sticky top-0 z-50 flex items-center justify-between px-6 py-4 bg-white dark:bg-gray-900 shadow">
      <Link href="/" className="flex items-center space-x-2" aria-label="Home">
        <span className="text-2xl font-bold tracking-tight text-gray-400">
          {"<"}
          <span className="text-green-500">Jp</span>
          {" />"}
        </span>
      </Link>
      <ul className="flex space-x-8 font-medium">
        {navLinks.map(link => {
          const isActive = pathname === link.href;
          return (
            <li key={link.href}>
              <Link
                href={link.href}
                className={`
                  group relative transition-colors duration-200
                  text-gray-700 dark:text-gray-200 hover:text-gray-100
                `}
              >
                {link.label}
                <span
                  className={`
                    pointer-events-none absolute left-0 -bottom-1 h-[2px] w-full
                    transition-all duration-300 rounded
                    bg-gray-100
                    ${isActive ? "scale-x-100" : "scale-x-0 group-hover:scale-x-100"}
                    origin-left
                  `}
                  aria-hidden="true"
                />
              </Link>
            </li>
          );
        })}
      </ul>
    </nav>
  );
}