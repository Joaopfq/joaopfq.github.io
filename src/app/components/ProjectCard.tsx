"use client";

import { motion } from "framer-motion";
import { FaExternalLinkAlt } from "react-icons/fa";
import { FaGithub } from "react-icons/fa";
import Link from "next/link";

interface ProjectCardProps {
  title: string;
  description: string;
  techStack?: string[];
  githubLink: string;
  liveLink?: string;
}

export function ProjectCard({ title, description, techStack, githubLink, liveLink }: ProjectCardProps) {
  return (
    <motion.div
      onClick={() => {}}
      whileHover={{ scale: 1.03, y: -5 }}
      transition={{ type: "spring", stiffness: 300 }}
      className="bg-white dark:bg-gray-700 rounded-xl shadow-md p-6 flex flex-col justify-between h-full hover:shadow-lg"
    >
      <Link href={githubLink} target="_blank" rel="noopener noreferrer" className="flex flex-col h-full">
        <div>
          <h4 className="text-xl font-semibold mb-2 text-gray-900 dark:text-white">{title}</h4>
          <p className="text-sm text-gray-700 dark:text-gray-300 mb-4">{description}</p>

          {techStack?.length && (
            <div className="mb-4">
              <span className="font-medium text-gray-800 dark:text-gray-200">Tech Stack:</span>
              <ul className="list-disc list-inside text-sm text-gray-700 dark:text-gray-300 mt-1">
                {techStack.map((tech, index) => (
                  <li key={index}>{tech}</li>
                ))}
              </ul>
            </div>
          )}
        </div>

      </Link>
        <div className="mt-2 space-x-4 flex">
          {liveLink && (
            <a
              className=" text-gray-500 hover:text-gray-700 dark:text-gray-300 dark:hover:text-gray-100 transition-colors duration-200"
              href={liveLink}
              target="_blank"
              rel="noopener noreferrer"
              aria-label="Acessar site"
            >
              <FaExternalLinkAlt size={22} />
            </a>
          )}
          <a
            className="text-gray-500 hover:text-gray-700 dark:text-gray-300 dark:hover:text-gray-100 transition-colors duration-200"
            href={githubLink}
            target="_blank"
            rel="noopener noreferrer"
            aria-label="GitHub"
          >
            <FaGithub size={25} />
          </a>
        </div>
    </motion.div>
  );
}
