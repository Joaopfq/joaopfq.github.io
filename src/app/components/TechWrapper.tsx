import React from 'react'
import { FaReact } from "react-icons/fa";
import { SiNextdotjs, SiTailwindcss, SiTypescript, SiVuedotjs } from 'react-icons/si';


type Tech = {
  name: string;
  icon: React.ReactNode;
  colorClass: string;
};

function TechWrapper() {

    const techs: Tech[] = [
      { name: "React", icon: <FaReact />, colorClass: "text-cyan-400" },
      { name: "Next.js", icon: <SiNextdotjs />, colorClass: "text-black dark:text-white" },
      { name: "Vue.js", icon: <SiVuedotjs />, colorClass: "text-green-500" },
      { name: "TypeScript", icon: <SiTypescript />, colorClass: "text-blue-600" },
      { name: "Tailwind CSS", icon: <SiTailwindcss />, colorClass: "text-teal-400" },
    ];

  return (
    <div className="flex flex-wrap justify-center gap-6">
        {techs.map((tech) => (
          <div
            key={tech.name}
            className="flex flex-col items-center p-4 rounded-lg shadow bg-white dark:bg-gray-700 hover:scale-105 transition-transform w-28"
          >
            <span className={`text-4xl mb-2 ${tech.colorClass}`}>{tech.icon}</span>
            <span className="text-sm font-medium text-center text-gray-700 dark:text-gray-300">{tech.name}</span>
          </div>
        ))}
      </div>
  )
}

export default TechWrapper