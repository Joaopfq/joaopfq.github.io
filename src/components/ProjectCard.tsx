"use client";

import { motion } from "framer-motion";
import { FaExternalLinkAlt } from "react-icons/fa";
import { FaGithub } from "react-icons/fa";
import Link from "next/link";
import Image, { StaticImageData } from "next/image";

interface TechStackIcon {
  icon: React.ReactNode;
  colorClass?: string;
  name: string;
}

interface ProjectCardProps {
  title: string;
  description: string;
  image?: StaticImageData;
  techStack?: TechStackIcon[];
  githubLink: string;
  liveLink?: string;
}

export function ProjectCard({ title, description, image, techStack, githubLink, liveLink }: ProjectCardProps) {
  return (
    <motion.div
      onClick={() => {}}
      whileHover={{ scale: 1.02, y: -3 }}
      transition={{ type: "spring", stiffness: 300 }}
      className="bg-neutral-900 rounded-xl shadow-md flex flex-col justify-between h-full hover:shadow-lg group"
    >
      <Link href={githubLink} target="_blank" rel="noopener noreferrer" className="flex flex-col w-full h-full">
        <div>
          <div
            className="
              align-center
              bg-zinc-950
              transition-colors
              duration-300
              p-4 mb-2 rounded-t-xl
              group-hover:bg-green-700
              group-hover:text-gray-100
            "
          >
            <h4 className="text-xl font-semibold pl-4 text-gray-100 group-hover:text-gray-100 transition-colors duration-300">
              {title}
            </h4>
          </div>
          <div>
            {image && (
              <div className="w-full h-40 relative mb-4">
                <Image
                  src={image}
                  alt={`${title} preview`}
                  fill
                  className="object-cover rounded-lg"
                  sizes="(max-width: 768px) 100vw, 400px"
                  priority={false}
                />
              </div>
            )}
            <div className="px-6" >
              <p className="text-sm text-gray-200 mb-4">{description}</p>
              {techStack && techStack.length > 0 && (
                <div className="mb-4 flex items-center gap-3 flex-wrap">
                  {techStack.map((tech, idx) =>
                    tech.icon ? (
                      <span key={idx} className={`text-2xl ${tech.colorClass}`} title={tech.name}>
                        {tech.icon}
                      </span>
                    ) : null
                  )}
                </div>
              )}
            </div>
          </div>
        </div>
      </Link>
      <div className="p-6 flex gap-4">
        {liveLink && (
          <a
            className="flex items-center gap-2 bg-foreground/90 hover:bg-foreground text-gray-800 px-4 py-2 rounded-lg font-medium transition-colors duration-200"
            href={liveLink}
            target="_blank"
            rel="noopener noreferrer"
            aria-label="Acessar site"
          >
            <FaExternalLinkAlt size={20} />
            <span>Demo</span>
          </a>
        )}
        <a
          className="flex items-center gap-2 bg-background hover:bg-background/80 text-gray-100 px-4 py-2 rounded-lg font-medium transition-colors duration-200"
          href={githubLink}
          target="_blank"
          rel="noopener noreferrer"
          aria-label="GitHub"
        >
          <FaGithub size={22} />
          <span>GitHub</span>
        </a>
      </div>
    </motion.div>
  );
}