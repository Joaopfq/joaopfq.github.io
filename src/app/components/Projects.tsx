import { ProjectCard } from "./ProjectCard";

const projects = [
  {
    title: "Financi Web",
    description: "Aplicação web para controle financeiro pessoal através do da definição de metas e gerenciamento de entradas e saidas",
    githubLink: "https://github.com/grupo-bala/financi-web-front-end",
    techStack: ["Vue", "TypeScript", "PostgreSQL"],
  },
  {
    title: "Financi Desktop",
    description: "Aplicação desktop para controle financeiro pessoal através do da definição de metas e gerenciamento de entradas e saidas",
    githubLink: "https://github.com/grupo-bala/financi-desktop",
    techStack: ["Java", "JavaFX", "PostgreSQL"],
  },
  {
    title: "Birdie",
    description: "App social desenvolvido com Next.js, React, TypeScript e Tailwind.css",
    githubLink: "https://github.com/Joaopfq/birdie",
    techStack: ["Next.js", "React", "TypeScript", "Tailwind.css"],
    liveLink: "https://birdie-omega.vercel.app/"
  },
  {
    title: "Track Pet",
    description: "Aplicação web social para reportar pets encontrados e perdidos.",
    githubLink: "https://github.com/Joaopfq/track-pet",
    techStack: ["Next.js", "React", "TypeScript", "Tailwind.css"],
  },
  {
    title: "MangaStore",
    description: "Aplicação web para gerenciamento de mangás",
    githubLink: "https://github.com/Joaopfq/MangaStore",
    techStack: ["Vue", "TypeScript", "Java", "Spring Boot"],
  },
  {
    title: "TERTS",
    description: "Aplicação web para ensino de refatoração de código de teste",
    githubLink: "https://github.com/DeigelaLima/Tutor_de_ensino_TS",
    techStack: ["Vue", "TypeScript", "Node.js", "JavaScript"],
    liveLink: "https://tutor-de-ensino-ts.vercel.app/"
  },
];

export function Projects() {
    return (
      <div id="projetos" className="py-16 px-4 bg-gray-100 dark:bg-gray-800">
        <h3 className="text-3xl font-bold text-center mb-8 text-gray-900 dark:text-white">Projetos</h3>
        <div className="grid gap-6 sm:grid-cols-2 lg:grid-cols-3 max-w-6xl mx-auto">
          {projects.map((project, index) => (
            <ProjectCard
              key={index}
              title={project.title}
              description={project.description}
              techStack={project.techStack}
              githubLink={project.githubLink}
              liveLink={project.liveLink}
            />
          ))}
        </div>
      </div>
    );
  }
  
  