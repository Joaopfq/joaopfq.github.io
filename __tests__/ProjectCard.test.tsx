import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { ProjectCard } from '@/app/components/ProjectCard';

describe('ProjectCard', () => {
  const mockProps = {
    title: 'Test Project',
    description: 'This is a test project.',
    techStack: ['React', 'TypeScript', 'Tailwind CSS'],
    githubLink: 'https://github.com/example/test-project',
    liveLink: 'https://example.com',
  };

  it('renders the title and description', () => {
    render(<ProjectCard {...mockProps} />);
    expect(screen.getByText('Test Project')).toBeInTheDocument();
    expect(screen.getByText('This is a test project.')).toBeInTheDocument();
  });

  it('renders the tech stack when provided', () => {
    render(<ProjectCard {...mockProps} />);
    expect(screen.getByText('Tech Stack:')).toBeInTheDocument();
    expect(screen.getByText('React')).toBeInTheDocument();
    expect(screen.getByText('TypeScript')).toBeInTheDocument();
    expect(screen.getByText('Tailwind CSS')).toBeInTheDocument();
  });

  it('does not render the tech stack when none is provided', () => {
    // Pass props without the techStack field
    const propsWithoutTechStack = {
      ...mockProps,
      techStack: undefined, // Ensure techStack is not provided
    };
    render(<ProjectCard {...propsWithoutTechStack} />);
    expect(screen.queryByText('Tech Stack:')).not.toBeInTheDocument();
  });

  it('renders the GitHub link with the correct href', () => {
    render(<ProjectCard {...mockProps} />);
    const githubLink = screen.getByLabelText('GitHub');
    expect(githubLink).toBeInTheDocument();
    expect(githubLink).toHaveAttribute('href', mockProps.githubLink);
    expect(githubLink).toHaveAttribute('target', '_blank');
    expect(githubLink).toHaveAttribute('rel', 'noopener noreferrer');
  });

  it('renders the live link when provided', () => {
    render(<ProjectCard {...mockProps} />);
    const liveLink = screen.getByLabelText('Acessar site');
    expect(liveLink).toBeInTheDocument();
    expect(liveLink).toHaveAttribute('href', mockProps.liveLink);
    expect(liveLink).toHaveAttribute('target', '_blank');
    expect(liveLink).toHaveAttribute('rel', 'noopener noreferrer');
  });

  it('does not render the live link when none is provided', () => {
    // Pass props without the liveLink field
    const propsWithoutLiveLink = {
      ...mockProps,
      liveLink: undefined, // Ensure liveLink is not provided
    };
    render(<ProjectCard {...propsWithoutLiveLink} />);
    expect(screen.queryByText('Acessar site')).not.toBeInTheDocument();
  });

});