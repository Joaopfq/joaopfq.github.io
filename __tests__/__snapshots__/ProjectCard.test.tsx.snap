// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectCard matches snapshot 1`] = `
<div>
  <div
    class="bg-white dark:bg-gray-700 rounded-xl shadow-md p-6 flex flex-col justify-between h-full hover:shadow-lg"
  >
    <div>
      <h4
        class="text-xl font-semibold mb-2 text-gray-900 dark:text-white"
      >
        Test Project
      </h4>
      <p
        class="text-sm text-gray-700 dark:text-gray-300 mb-4"
      >
        This is a test project.
      </p>
      <div
        class="mb-4"
      >
        <span
          class="font-medium text-gray-800 dark:text-gray-200"
        >
          Tech Stack:
        </span>
        <ul
          class="list-disc list-inside text-sm text-gray-700 dark:text-gray-300 mt-1"
        >
          <li>
            React
          </li>
          <li>
            TypeScript
          </li>
          <li>
            Tailwind CSS
          </li>
        </ul>
      </div>
    </div>
    <div
      class="mt-4 space-x-4"
    >
      <a
        class="text-blue-600 dark:text-blue-400 hover:underline font-medium"
        href="https://github.com/example/test-project"
        rel="noopener noreferrer"
        target="_blank"
      >
        GitHub
      </a>
      <a
        class="text-green-600 dark:text-green-400 hover:underline font-medium"
        href="https://example.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        Acessar site
      </a>
    </div>
  </div>
</div>
`;
